### PR TITLE
[Profiler] Add API for Dynamic Activity Toggling [2/n]

### PR DESCRIPTION
**Pull request by sraikund16**
Summary: During PT2 there are many GPU/CPU events that are unneccessary to profile in between a given step. To remedy this, we can add an API that takes in a list of activities and an arg to toggle said activies or not. For this diff we are adding the profiler API to propogate down to kineto (and in the future the collection.cpp logic). Subsequent diffs will be added for CPU toggling and e2e testing.

Test Plan: Tested by toggling backward gpu traces off and got following trace: https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/devvm2185.cco0.facebook.com/rank-0.Jul_31_13_40_55.3251726.pt.trace.json.gz&bucket=gpu_traces

Reviewed By: aaronenyeshi

Differential Revision: D60541767

### CODE REVIEW

**Review by aaronenyeshi — APPROVED** [review]

File: third_party/kineto
Comment (soulitzer) [inline third_party/kineto:1]: Just to double check, is commiting this submodule update intentional? I had imagined them to be usually done as part of independent PRs.

Comment (sraikund16) [reply third_party/kineto:1]: It is intentional, only because the difference between the current submodule and the new one is just the kineto side of this feature. This will help ensure that there are no issues in OSS if someone calls this API
